### PR TITLE
Add GolangCI action.

### LINKED ISSUE #708: GolangCI is closing, what to use now
**Topic**
A few days ago, GolangCI [announced ](https://medium.com/golangci/golangci-com-is-closing-d1fc1bd30e0e) that it is closing. This is of course very unfortunate as their service was very usefull in PR's. Now the question is, what are we gonna do with the GolangCI test we have running now?

**Your thoughts**
We can
 * Use github actions to check the code
 * Host GolangCI on a server
 * Switch to a different provider

### PR DESCRIPTION
**Pull request by sstiglitz**
This is to close #708.

The result of adding this linter ends up raising issues with the existing code base and you can see from this PR. They all are trivial fixes and I don't mind cleaning them up, but I wanted to check with you first before making changes to to the code base. Please let me know your thoughts.

### CODE REVIEW

**Comment by jesseduffield** [comment]
LGTM, could you take a look @dawidd6 ?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: golangci-lint
+on:
```
Comment (dawidd6) [inline .github/workflows/lint.yml:2]: I think a simple `on: pull_request` trigger will suffice.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+jobs:
+  golangci:
+    strategy:
+      matrix:
+        go-version: 
+          - 1.14.x
+        os: 
+          - ubuntu-latest
+    name: lint
+    runs-on: ${{ matrix.os }}
```
Comment (dawidd6) [inline .github/workflows/lint.yml:29]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  pull_request_review:
+    types:
+      - submitted
+  check_suite:
+    types:
+      - completed
+  status: {}
+jobs:
+  golangci:
+    strategy:
```
Comment (dawidd6) [inline .github/workflows/lint.yml:22]: No need for all of that.

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      matrix:
+        go-version: 
+          - 1.14.x
+        os: 
+          - ubuntu-latest
+    name: lint
```
Comment (dawidd6) [inline .github/workflows/lint.yml:28]:
```suggestion
```

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
```
Comment (dawidd6) [inline .github/workflows/lint.yml:32]:
```suggestion
```

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        uses: golangci/golangci-lint-action@v2
```
Comment (dawidd6) [inline .github/workflows/lint.yml:33]:
```suggestion
      - uses: golangci/golangci-lint-action@v2
```

File: .github/workflows/lint.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+name: golangci-lint
```
Comment (dawidd6) [inline .github/workflows/lint.yml:1]:
```suggestion
name: Lint
```

**Review by dawidd6 — CHANGES_REQUESTED** [review]

**Review by dawidd6 — APPROVED** [review]

**Comment by dawidd6** [comment]
Thanks!

**Review by jesseduffield — APPROVED** [review]
LGTM

**Pull request merged**